ListingManager.Tests/ListingInformationTests.cs

```diff
@@ -1,45 +1,241 @@
+            "    public class Program { }",
+            "}"
+        };
+
+        DirectoryInfo tempDir = CreateTempDirectory(new(Path.GetTempPath()));
+        WriteFiles(tempDir, filesToMake, toWrite);
+        var writtenFiles = WriteFiles(tempDir, filesToMake, toWrite);
+        Assert.Single(writtenFiles);
 
-        Assert.AreEqual(chapterNumber, listingInformation.ChapterNumber);
-        Assert.AreEqual(listingNumber, listingInformation.ListingNumber);
+        ListingInformation listingInformation = new(writtenFiles.First().FullName);
 
-        Assert.AreEqual(suffix ?? "", listingInformation.ListingSuffix);
+        Assert.Equal(chapterNumber, listingInformation.OriginalChapterNumber);
+        Assert.Equal(listingInformation.NewChapterNumber, listingInformation.OriginalChapterNumber);
 
-        Assert.AreEqual(description ?? "", listingInformation.ListingDescription);
+        Assert.Equal(listingNumber, listingInformation.OriginalListingNumber);
+        Assert.Equal(listingInformation.NewListingNumber, listingInformation.OriginalListingNumber);
+
+        Assert.Equal(suffix, listingInformation.OriginalListingNumberSuffix);
+        Assert.Equal(listingInformation.NewListingNumberSuffix, listingInformation.OriginalListingNumberSuffix);
+
+        Assert.Equal(caption, listingInformation.Caption);
+
+        Assert.False(listingInformation.Changed);
     }
 
-    [TestMethod]
-    [DataRow("Listing01.01.cs")]
-    [DataRow("Listing05.04.Something.xml")]

-    [DataRow("Listing05.04.Something.XML")]

+    [Theory]
+    [InlineData("Listing01.01.cs")]
+    [InlineData("Listing05.04.Something.xml")]
+    [InlineData("Listing05.04.Something.XML")]
     public void Constructor_GivenValidListingFileTypes_CreatesNewListingInformation(string listing)
     {
-        ListingInformation listingInformation = new(listing);

-        Assert.IsNotNull(listingInformation);

-        Assert.AreEqual(System.IO.Path.GetExtension(listing), listingInformation.ListingExtension);

+        List<string> filesToMake = new()
+        {
+            listing
+        };
+
+        IEnumerable<string> toWrite = new List<string>
+        {
+            "namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter18.Listing18_01",
+            "{",
+            "    using System;",
+            "    using System.Reflection;",
+            "    public class Program { }",
+            "}"
+        };
+
+        DirectoryInfo tempDir = CreateTempDirectory(new(Path.GetTempPath()));
+        WriteFiles(tempDir, filesToMake, toWrite);
+        var writtenFiles = WriteFiles(tempDir, filesToMake, toWrite);
+        Assert.Single(writtenFiles);
+
+        ListingInformation listingInformation = new(writtenFiles.First().FullName);

+        Assert.NotNull(listingInformation);

+        Assert.Equal(Path.GetExtension(listing), listingInformation.ListingExtension);
+        Assert.False(listingInformation.Changed);
+    }

    [Theory]

```

Comment:
```suggestion
    }
    
    [Theory]

```

Reply:
Something weird is happening with these, you marked one of these before as well but it looked normal in VS

![image](https://github.com/IntelliTect/EssentialCSharp.ListingManager/assets/22186029/89f1e6af-a777-4cfd-b26d-968c1e76a7ee)

Reply:
Turn on witespace, you have a character there that is not a normal space.